PowerEditor/src/WinControls/shortcut/shortcut.cpp

```diff
@@ -168,10 +168,10 @@ generic_string Shortcut::toString() const
 	return sc;
 }
 
-void Shortcut::setName(const TCHAR * name)
+void Shortcut::setName(const TCHAR * menuName, const TCHAR * shortcutName)
 {
-	lstrcpyn(_menuName, name, nameLenMax);
-	lstrcpyn(_name, name, nameLenMax);
+	lstrcpyn(_menuName, menuName, nameLenMax);
+	TCHAR const * name = shortcutName ? shortcutName : menuName;

```

Comment:
Shouldn't it be `const TCHAR *` instead of `TCHAR const *` ?

Reply:
This may be an English speaker thing, but...
If you have, `TCHAR const * foo`, you read it (RTL) out loud as "foo is a pointer to constant TCHAR".
Putting the `const` in a different place, it doesn't read so nice.
THAT is why I like it this way.
But, does it really matter? :-)
That leads into things like, is this correct: `TCHAR *foo` or is this correct: `TCHAR* foo`... or should the `*` be jammed up against the variable name, the type, or floating in space (like line 174 has it above)?
But...these are "small" things, and I'm sure throughout the N++ code we have a variety of all of them. :-)

Reply:
Both should have the same effect, see

https://stackoverflow.com/questions/890535/what-is-the-difference-between-char-const-and-const-char

From this:
> The first thing to the left of the "const" is what's constant. If "const" is the thing the farthest to the left, then the first thing to the right of it is what's constant

> To avoid confusion, always append the const qualifier.
int       *      mutable_pointer_to_mutable_int;
int const *      mutable_pointer_to_constant_int;
int       *const constant_pointer_to_mutable_int;
int const *const constant_pointer_to_constant_int;

The **const** itself is necessary, because the parameter is **const**. This **const** prevents me from writing `menuName[3]='7'` wihin this function. The **const** has to go over to **name** to satisfy the very accurate compiler, so I appended **const** out of habit. In most of the code lines of Notepad++ **const** seems to be first in the line, and I may have obeyed this in the past, so if you want, I swap it.

Reply:
>  is this correct: `TCHAR *foo` or is this correct: `TCHAR* foo`... or

I admit that I act totally random about spaces around the `*`. I use the way it is written at the nearest point in code I see, and this not in Notepad++ only. Although at 1st glance, the 1st version seems more correct because of `THCAR *foo, foo1, *foo2;`

Reply:
>I admit that I act totally random about spaces around the *

Yes, use what the code around the area you are affecting does.
However, I've noticed that Visual Studio has its own opinion on this, and sometimes changes it on you.

Reply:
> Visual Studio has its own opinion on this, and sometimes changes it on you.

Does it? I never noticed. When?

Reply:
@sasumner @Uhf7 OK, thank you for the explanation. I didn't know both are the same semantically.
Let's keep it as it is.